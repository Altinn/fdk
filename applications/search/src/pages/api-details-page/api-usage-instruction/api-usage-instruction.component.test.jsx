import React from 'react';

import { shallow } from 'enzyme';

import ApiUsageInstruction from './api-usage-instruction.component';

import apiItem from './__fixtures/apiItem.json';

test('should render api-ApiUsageInstruction correctly with no props', () => {
  expect(shallow(<ApiUsageInstruction />)).toMatchSnapshot();
});

test('should render api-ApiUsageInstruction correctly', () => {
  const {
    openApi: { servers },
    apiSpecUrl,
    apiDocUrl
  } = apiItem;
  expect(
    shallow(
      <ApiUsageInstruction
        servers={servers}
        apiSpecUrl={apiSpecUrl}
        apiDocUrl={apiDocUrl}
      />
    )
  ).toMatchSnapshot();
});

test('should render api-ApiUsageInstruction correctly apiSpecUrl is not set', () => {
  const {
    openApi: { servers },
    apiDocUrl
  } = apiItem;
  expect(
    shallow(<ApiUsageInstruction servers={servers} apiDocUrl={apiDocUrl} />)
  ).toMatchSnapshot();
});

test('should render api-ApiUsageInstruction correctly apiSpecUrl is not set', () => {
  const {
    openApi: { servers },
    apiSpecUrl
  } = apiItem;
  expect(
    shallow(<ApiUsageInstruction servers={servers} apiSpecUrl={apiSpecUrl} />)
  ).toMatchSnapshot();
});

test('should render api-ApiUsageInstruction correctly apiSpecUrl and apiDocUrl are not set', () => {
  const {
    openApi: { servers }
  } = apiItem;
  expect(shallow(<ApiUsageInstruction servers={servers} />)).toMatchSnapshot();
});
