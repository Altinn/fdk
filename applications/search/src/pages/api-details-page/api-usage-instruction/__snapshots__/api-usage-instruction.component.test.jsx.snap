// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render api-ApiUsageInstruction correctly 1`] = `
<ListRegular
  bottomMargin={true}
  title="Start using the API"
>
  <div
    className="d-flex list-regular--item"
  >
    <div
      className="col-4 pl-0"
    >
      <h4
        className=""
      >
        Server URL
      </h4>
    </div>
    <div
      className="col-8"
    >
      <h4
        className=""
      >
        Description
      </h4>
    </div>
  </div>
  <TwoColRow
    col1="https://fellesdatakatalog.brreg.no/api-cat/apis"
    col2="The production API server"
    key="https://fellesdatakatalog.brreg.no/api-cat/apisThe production API server"
  />
  <div
    className="d-flex list-regular--item"
  >
    <div
      className="col-12 pl-0"
    >
      <a
        className="mr-5"
        href="https://informasjonsforvaltning.github.io/felles-datakatalog/api-katalog/api/"
      >
        Documentation
        <i
          className="fa fa-external-link fdk-fa-right"
        />
      </a>
      <a
        className="mr-5"
        href="https://raw.githubusercontent.com/brreg/openAPI/master/specs/api-cat.json"
      >
        Specification
        <i
          className="fa fa-external-link fdk-fa-right"
        />
      </a>
    </div>
  </div>
</ListRegular>
`;

exports[`should render api-ApiUsageInstruction correctly apiSpecUrl and apiDocUrl are not set 1`] = `
<ListRegular
  bottomMargin={true}
  title="Start using the API"
>
  <div
    className="d-flex list-regular--item"
  >
    <div
      className="col-4 pl-0"
    >
      <h4
        className=""
      >
        Server URL
      </h4>
    </div>
    <div
      className="col-8"
    >
      <h4
        className=""
      >
        Description
      </h4>
    </div>
  </div>
  <TwoColRow
    col1="https://fellesdatakatalog.brreg.no/api-cat/apis"
    col2="The production API server"
    key="https://fellesdatakatalog.brreg.no/api-cat/apisThe production API server"
  />
</ListRegular>
`;

exports[`should render api-ApiUsageInstruction correctly apiSpecUrl is not set 1`] = `
<ListRegular
  bottomMargin={true}
  title="Start using the API"
>
  <div
    className="d-flex list-regular--item"
  >
    <div
      className="col-4 pl-0"
    >
      <h4
        className=""
      >
        Server URL
      </h4>
    </div>
    <div
      className="col-8"
    >
      <h4
        className=""
      >
        Description
      </h4>
    </div>
  </div>
  <TwoColRow
    col1="https://fellesdatakatalog.brreg.no/api-cat/apis"
    col2="The production API server"
    key="https://fellesdatakatalog.brreg.no/api-cat/apisThe production API server"
  />
  <div
    className="d-flex list-regular--item"
  >
    <div
      className="col-12 pl-0"
    >
      <a
        className="mr-5"
        href="https://informasjonsforvaltning.github.io/felles-datakatalog/api-katalog/api/"
      >
        Documentation
        <i
          className="fa fa-external-link fdk-fa-right"
        />
      </a>
    </div>
  </div>
</ListRegular>
`;

exports[`should render api-ApiUsageInstruction correctly apiSpecUrl is not set 2`] = `
<ListRegular
  bottomMargin={true}
  title="Start using the API"
>
  <div
    className="d-flex list-regular--item"
  >
    <div
      className="col-4 pl-0"
    >
      <h4
        className=""
      >
        Server URL
      </h4>
    </div>
    <div
      className="col-8"
    >
      <h4
        className=""
      >
        Description
      </h4>
    </div>
  </div>
  <TwoColRow
    col1="https://fellesdatakatalog.brreg.no/api-cat/apis"
    col2="The production API server"
    key="https://fellesdatakatalog.brreg.no/api-cat/apisThe production API server"
  />
  <div
    className="d-flex list-regular--item"
  >
    <div
      className="col-12 pl-0"
    >
      <a
        className="mr-5"
        href="https://raw.githubusercontent.com/brreg/openAPI/master/specs/api-cat.json"
      >
        Specification
        <i
          className="fa fa-external-link fdk-fa-right"
        />
      </a>
    </div>
  </div>
</ListRegular>
`;

exports[`should render api-ApiUsageInstruction correctly with no props 1`] = `
<ListRegular
  bottomMargin={true}
  title="Start using the API"
>
  <div
    className="d-flex list-regular--item"
  >
    <div
      className="col-4 pl-0"
    >
      <h4
        className=""
      >
        Server URL
      </h4>
    </div>
    <div
      className="col-8"
    >
      <h4
        className=""
      >
        Description
      </h4>
    </div>
  </div>
</ListRegular>
`;
